### ISSUE TITLE
Autocompletions don't update on keypress, causing missing completions

### ISSUE THREAD

**Issue by joehoyle** [tomv564/LSP#299]
The `on_query_completions` API/method is only called when Sublime wants to trigger the autocompletions, not mid-typing in the completion. Due to their being a `maxItems` on the LSP (which I think is normal) this causes missing completions.

If I type `w` I'll get all completions for `w` via `on_query_completions`, however this list is 100 items long. Assuming I have a lot of completions for `w`, when I type `p` to form `wp`, `on_query_completions` is _not_ called (because I think that's how Sublime's API functions) as there are still _some_ `wp` items in the completion list. However, this has the potential to miss other `wp` completions that were not returned from the LSP server in the initial `/completion` request for `w`.

There really has, I think, to be one of 2 possilble fixes:

1. Somehow get `on_query_completions` called on every keypress once it has been triggered by sublime, thus being able to do a fresh query on the LSP server on each modification.
2. The LSP server should have no limit of the completions it returns, so sending `w` would return a _lot_ of functions (in my case due to indexing WordPress). I'd imagine this would have performance implications, but maybe is the correct thing to do.

**Comment by tomv564** [comment]
What you are describing is probably LSP not observing the `isIncomplete` field in the completion response. Can you confirm if the language server you are using returns `false` in this field when it sends a partial result set for `w`?

If we can set up a clean way to handle successive keypresses within a completion, then nothing is stopping us from refreshing the completions on those keypresses.

**Comment by joehoyle** [comment]
@tomv564 I can confirm the server is returning `isIncomplete: true` in the response:

```
LSP:      {'isIncomplete': True, 'items': [{'kind': 14, ...
```

I've looked around quite a bit to better understand the expectations of `on_query_completions` in terms of further typing - as far as I can see, it's not supported. Doing something like `CompletionHandler::on_modified` to hide / show the completions may be an option?

**Comment by tomv564** [comment]
Nice, sounds like the server is just waiting for us to do the right thing :)

The `on_modified` function should get called for every keypress, so with the right conditions you could refresh completions from there!

**Comment by kadircet** [comment]
Any updates regarding this one? It looks like completion caching mechanism still does not respect isIncomplete field in server's response, which results in bad code completion quality.

https://github.com/tomv564/LSP/blob/master/plugin/completion.py#L219

**Comment by tomv564** [comment]
No update yet, you should see a linked PR against this issue if someone starts on it!

**Comment by rwols** [comment]
should be fixed by #611
